powershell/resources/psruntime/BuildTime/Cmdlets/NewAssemblyInfo.cs

```diff
@@ -0,0 +1,64 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+using System;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Rest.ClientRuntime.PowerShell
+{
+    [Cmdlet("New", "AssemblyInfo")]
+    [DoNotExport]
+    public class NewAssemblyInfo : PSCmdlet
+    {
+        private readonly string assemblyInfoPath = Path.Combine("${$project.baseFolder}", "Properties", "AssemblyInfo.cs");
+        private const string assemblyName = "${$project.title}";
+        private const string assemblyVersion = "${$project.moduleVersion}";
+        private const string assemblyCompanyName = "${$project.assemblyCompany}";
+        private const string assemblyProduct = "${$project.assemblyProduct}";
+        private const string assemblyCopyright = "${$project.assemblyCopyright}";
+        protected override void ProcessRecord()
+        {
+            try
+            {
+                if (File.Exists(assemblyInfoPath))
+                {
+                    return;
+                }
+                StringBuilder sb = new StringBuilder();
+                sb.AppendLine(@"
+# ----------------------------------------------------------------------------------
+${$project.pwshCommentHeaderForCsharp}
+# ----------------------------------------------------------------------------------
+");
+                sb.Append($"{Environment.NewLine}");
+                sb.AppendLine("using System;");
```

Comment:
We could consider hardcode all the stuff except for assemblyName, Guid, and assemblyVersion.